packages/rich-text/src/component/index.native.js

```diff
@@ -1070,24 +1079,48 @@ const withFormatTypes = ( WrappedComponent ) => ( props ) => {
 
 export default compose( [
 	withSelect( ( select, { clientId } ) => {
-		const { getBlockParents, getBlock, getSettings } = select(
-			'core/block-editor'
-		);
+		const {
+			getBlockParents,
+			getBlock,
+			getSettings,
+			getBlockAttributes,
+		} = select( 'core/block-editor' );
 		const parents = getBlockParents( clientId, true );
 		const parentBlock = parents ? getBlock( parents[ 0 ] ) : undefined;
 		const parentBlockStyles = get( parentBlock, [
 			'attributes',
 			'childrenStyles',
 		] );
-		const baseGlobalStyles = getSettings()
-			?.__experimentalGlobalStylesBaseStyles;
+
+		const settings = getSettings();
+
+		const baseGlobalStyles = settings?.__experimentalGlobalStylesBaseStyles;
+		const fontSizes =
+			settings?.__experimentalFeatures?.typography?.fontSizes?.theme ??
+			settings.fontSizes ??
+			[];
+
+		const currentBlockAttributes = getBlockAttributes( clientId );
+		const blockFontSize = currentBlockAttributes?.fontSize;
+		const blockStyle = currentBlockAttributes?.style;
+
+		const fontSizeObject = getBlockFontSize(
+			fontSizes,
+			blockFontSize,
+			blockStyle?.typography?.fontSize
+		);
+		const blockFontSizeValue =
+			fontSizeObject?.size ||
+			blockStyle?.typography?.fontSize ||
+			blockFontSize;
```

Comment:
I wonder if we should add this in a separate PR, this way this one would be mainly the UI what do you think?

Reply:
Yes, I will remove this part since it doesn't quite work as expected.